sw/device/boot_rom2/final_jump.S

```diff
@@ -0,0 +1,91 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+  /**
+   * Linker-provided symbols for stack layout information.
+   */
+  .extern _stack_start
+  .extern _stack_size
+
+_final_jump_asm:
+  .globl _final_jump_asm
+
+  // Bleach the ROM stack, which we no longer need for anything.
+  // Here, |t0| is the current word being zeroed, while |t1| is
+  // the number of bytes left to zero.
+  la   t0, _stack_start
+  la   t1, _stack_size
+1:
+  addi t0, t0, -4
+  addi t1, t1, -4
+  sw   zero, 0(t0)
+  bnez t1, 1b
+
+  // Clobber every register, including |ra| and |sp|. However, do
+  // not clobber |a0| and |a1|, which contain important information.
```

Comment:
What important information? Presumably `a0` is the start of the new code, and `a1` is `next_stage_args_t` - this needs to be documented, preferably at function entry (I know it's following the RISC-V ABI, but having them explicitly documented makes this much easier to review)

Reply:
Done.